examples/i18n/main.rs

```diff
@@ -0,0 +1,117 @@
+use std::collections::HashMap;
+use std::env;
+use std::fs::read_to_string;
+use clap::text_provider::TextProvider;
+use rust_i18n::set_locale;
+use rust_i18n::t;
+use rust_i18n::Backend;
+use std::path::PathBuf;
+use clap::{Parser, Subcommand};
+
+pub struct RemoteI18n {
+    pub trs: HashMap<String, HashMap<String, String>>,
+}
+
+impl RemoteI18n {
+    fn new() -> Self {
+        let en = read_to_string("./examples/i18n/locales/en.yml").unwrap();
+        let jp = read_to_string("./examples/i18n/locales/jp.yml").unwrap();
+        let en_trs = serde_yml::from_str::<HashMap<String, String>>(&en).unwrap();
+        let jp_trs = serde_yml::from_str::<HashMap<String, String>>(&jp).unwrap();
+        let trs = HashMap::from([("en".to_owned(), en_trs), ("jp".to_owned(), jp_trs)]);
+
+        Self {
+            trs
+        }
+    }
+}
+
+impl Backend for RemoteI18n {
+    fn available_locales(&self) -> Vec<&str> {
+        self.trs.keys().map(|k| k.as_str()).collect()
+    }
+
+    fn translate(&self, locale: &str, key: &str) -> Option<&str> {
+        self.trs.get(locale)?.get(key).map(|k| k.as_str())
+    }
+}
+
+rust_i18n::i18n!("locales", backend = RemoteI18n::new());
+
+
+#[derive(Parser)]
+#[command(version, about, long_about = None)]
+struct Cli {
+    /// {name-help-text}
+    #[arg(value_name = "{name}")]
+    name: Option<String>,
+
+    /// {config-help-text}
+    #[arg(short, long, value_name = "{file}")]
+    config: Option<PathBuf>,
+
+    /// {debug-help-text}
+    #[arg(short, long, action = clap::ArgAction::Count)]
+    debug: u8,
+
+    #[command(subcommand)]
+    command: Option<Commands>,
+}
+
+#[derive(Subcommand)]
+enum Commands {
+    /// {test-help-text}
+    Test {
+        /// {list.help-text}
+        #[arg(short, long)]
+        list: bool,
+    },
+}
+
+struct I18n;
+
+impl TextProvider for I18n {
+    fn get(&self, key: &str) -> impl AsRef<str> {
+        t!(key)
+    }
+}
+
+fn main() {
+    let locale = env::var("LOCALE").unwrap_or("en".into());
+    set_locale(&locale);
+    let cli = Cli::parse_with_texts(&I18n);
+    // You can check the value provided by positional arguments, or option arguments
+    if let Some(name) = cli.name.as_deref() {
+        println!("Value for name: {name}");
```

Comment:
The user would be free to continue using rust-i18n to internationalize this and the following printlines as well.